<a href="https://colab.research.google.com/github/MikeDeecode/KC-properties-price-analysis/blob/master/kc_properties_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**KC HOUSE SALES DATA**

**The aim of this project:**

* To detemine the features that affects the price of the house 
* How well the features affect the price of the house 
* Develop a regression model to predict the prices of new houses 
* To obtain valuable insights and improve the pricing of properties 

##**Exploratory Analysis**

Import the required libraries 

In [10]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns 
%matplotlib inline 

print("Setup complete")

Setup complete


In [11]:
kc_properties = pd.read_csv("/content/kc_house_data.csv")
print("Success")

Success


In [15]:
kc_properties.shape

(21613, 21)

* There are 21,613 rows (houses) and 21 columns (features) in this data 

In [12]:
kc_properties.head()

id             date     price  ...     long  sqft_living15  sqft_lot15
0  7129300520  20141013T000000  221900.0  ... -122.257           1340        5650
1  6414100192  20141209T000000  538000.0  ... -122.319           1690        7639
2  5631500400  20150225T000000  180000.0  ... -122.233           2720        8062
3  2487200875  20141209T000000  604000.0  ... -122.393           1360        5000
4  1954400510  20150218T000000  510000.0  ... -122.045           1800        7503

[5 rows x 21 columns]

**DESCRIPTION OF THE DATA**

**Below is a brief description of the columns in the data**

* **id** - the identification number of the house
* **date** - the date the house was sold 
* **price** - the predicted price of the house 
* **bedrooms** - th e number of bedrooms in the house
* **bathrooms** - the numbe rof bathrooms in the house 
* **sqft_living** - square footage of the house 
* **sqft_lot**- square footage of the parking lot 
* **floors** - total numbe rof floors in the house 
* **waterfront**- If the house has a waterfront view (1 for yes)
* **view** - how many times the house has been viewed 
* **condition** - How good the condition of the house is (5 is excellent)

In [14]:
kc_properties.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  